lib/xenops_server.ml

```diff
@@ -1504,42 +1521,65 @@ and perform ?subtask ?result (op: operation) (t: Xenops_task.t) : unit =
 			info "VM %s has memory_limit = %Ld" id state.Vm.memory_limit;
 
 			Open_uri.with_open_uri memory_url
-				(fun mfd ->
-					Sockopt.set_sock_keepalives mfd;
-					let open Xenops_migrate in
-					let module Request = Cohttp.Request.Make(Cohttp_posix_io.Unbuffered_IO) in
-					let cookies = [
-						"instance_id", instance_id;
-						"dbg", t.Xenops_task.dbg;
-						"memory_limit", Int64.to_string state.Vm.memory_limit;
-					] in
-					let headers = Cohttp.Header.of_list (
-						Cohttp.Cookie.Cookie_hdr.serialize cookies :: [
-							"Connection", "keep-alive";
-							"User-agent", "xenopsd";
-						]) in
-					let request = Cohttp.Request.make ~meth:`PUT ~version:`HTTP_1_1 ~headers memory_url in
-
-					Request.write (fun _ -> ()) request mfd;
-
-					begin match Handshake.recv mfd with
-						| Handshake.Success -> ()
-						| Handshake.Error msg ->
-							error "cannot transmit vm to host: %s" msg;
-							raise (Internal_error msg)
-					end;
+				(fun mem_mfd ->
```

Comment:
I think that `mfd` stood for "memory file-descriptor". So now it's a "memory memory file-descriptor"?